Engine/DataFeeds/DataManager.cs

```diff
@@ -48,16 +50,66 @@ public class DataManager : IAlgorithmSubscriptionManager, IDataFeedSubscriptionM
         public DataManager(
             IDataFeed dataFeed,
             UniverseSelection universeSelection,
-            IAlgorithmSettings algorithmSettings,
+            IAlgorithm algorithm,
             ITimeKeeper timeKeeper,
             MarketHoursDatabase marketHoursDatabase)
         {
             _dataFeed = dataFeed;
             UniverseSelection = universeSelection;
-            _algorithmSettings = algorithmSettings;
+            UniverseSelection.SetDataManager(this);
```

Comment:
I would typically perform this type of initialization from the call site as it would become more obvious if someone's setting it twice. Also, it's typically good to 'wire' up components outside of the component itself -- although, holy wars have been fought over the concepts of self-registration, but my position is that it's not the components job to decide it's registrations/wirings